From 0cb4ebc8d6c1fd1b30167639e65580936a520c8d Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solusos.com>
Date: Thu, 5 Sep 2013 12:06:59 +0100
Subject: [PATCH 1/3] pisi: Fix parsing of package update messages

Many packages in SolusOS now use markdown in the package update sections,
to allow more informative messages to be shown in gpk-update-viewer.
Multiline messages that use markdown bullet lists are broken due to incorrect
handling of \n characters (replace with ";").

This commit fixes the issue by replacing new lines with the ";" character
---
 backends/pisi/pisiBackend.py |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/backends/pisi/pisiBackend.py b/backends/pisi/pisiBackend.py
index c4d49e5..0df51dd 100755
--- a/backends/pisi/pisiBackend.py
+++ b/backends/pisi/pisiBackend.py
@@ -230,8 +230,7 @@ class PackageKitPisiBackend(PackageKitBaseBackend, PackagekitPackage):
                         update_release = int(update.getAttribute("release"))
                         updater = update.getTagData("Name")
                         update_message = update.getTagData("Comment")
-                        update_message = update_message.replace("\n\n", ";")
-                        update_message = update_message.replace("\n", " ")
+                        update_message = update_message.replace("\n", ";")
                         update_date = update.getTagData("Date")
                 return (update_message, update_date)
             pkg = pkg.nextTag("Package")
-- 
1.7.10.4

